app/models/office.rb

```diff
@@ -12,83 +14,105 @@ class Office < ApplicationRecord
 
   alias_attribute :to_s, :name
 
-  scope :with_claimed_shifts, -> { joins(needs: :shifts).merge(Shift.claimed) }
-  scope :with_claimed_needs, -> { joins(:needs).merge(Need.has_claimed_shifts) }
+  scope :with_claimed_shifts, -> (user) {
+    if user.admin?
+      joins(needs: :shifts).merge(Shift.claimed)
+    elsif user.coordinator? || user.social_worker?
+      joins(:office_users, needs: :shifts).where(office_users: { user: user }).merge(Shift.claimed)
+    else
+      raise 'You do not have the proper permissions'
+    end
+  }
+
+  scope :with_claimed_needs, -> (user) {
+    if user.admin?
+      joins(:needs).merge(Need.has_claimed_shifts)
+    elsif user.coordinator? || user.social_worker?
+      joins(:office_users, :needs).where(office_users: { user: user }).merge(Need.has_claimed_shifts)
+    else
+      raise 'You do not have the proper permissions'
+    end
+  }
 
   alias_attribute :to_s, :name
 
-  def self.claimed_shifts_by_office
-    with_claimed_shifts.group('offices.id')
+  def self.total_volunteer_hours_by_office(current_user, start_at, end_at)
+    with_claimed_shifts(current_user)
+      .then { |scope| filter_by_date_range(scope, start_at, end_at) } 
+      .group(:id, :name)
+      .sum('shifts.duration / 60.0')
```

Comment:
I feel like these methods that gather report data for `Office` and `User` don't really belong in the model. Seems like a service class would be more ideal, thoughts @benjaminwood?